src/nfa/nfa.cc

```diff
@@ -555,3 +555,36 @@ bool Nfa::is_identical(const Nfa& aut) const {
     }
     return delta == aut.delta;
 }
+
+Nfa& Nfa::unite_nondet_with(const mata::nfa::Nfa& aut) {
+    const size_t num_of_states{ this->num_of_states() };
+    const size_t aut_num_of_states{ aut.num_of_states() };
+    const size_t new_num_of_states{ num_of_states + aut_num_of_states };
+    auto renumber_states = [&](State st) {
+        return st + num_of_states;
+    };
+
+    if (this == &aut) {
+        throw std::runtime_error("Performing a useless in-place union on the same NFA object");
+    }
+
+    if (final.empty() || initial.empty()) { *this = aut; return *this; }
+    if (aut.final.empty() || aut.initial.empty()) { return *this; }
+
+    this->delta.reserve(new_num_of_states);
+    this->delta.allocate(num_of_states);
```

Comment:
What is this doing? Shouldn't delta already have `num_of_states` states? Or is this because of final and initial states that are not in delta?

Reply:
It is because of potential initial and final states which might not be allocated in the `Delta`. I will add a comment with this to improve readability for others.